### ISSUE TITLE
Untrusted devices not syncing/unexpected items

### ISSUE THREAD

**Issue by mklss** [syncthing/syncthing#7631]
### Does your log mention database corruption?
No

### Include required information

Versions 1.15.1 and 1.16.0.

### Problem

With untrusted devices, it's very easy to generate cases, where items are not synced (out of date, stalled forever), or where "unexpected items" pop up in the untrusted device.
The setup I have is two trusted devices A, B and one untrusted U, all fully connected.

- Adding new files seems to generally work.
- Changing files seems to very often case "out of date" and forever stalls on U.
- Sometimes, changing files (say on A) causes the "unexpected items" problem.
- Just play around a bit, and you should run into problems.

All in all, this setup is completely unusable.

### Suspected cause
Probably, the encrypted sync logic can't deal with A and B syncing, skipping U.
Perhaps you can confirm, that this is the case?
For now, I've removed the direct connection and will see how things play out.
If direct communication between A and B is not allowed/supported in this setting, then this should be mentioned in the docs.

Anyway, the encryption/untrusted device feature is central for so many use cases, I'm very happy that it's available.
Thanks for your work!

**Comment by imsodin** [comment]
As you mentioned v1.15.1: Did you do a full reset of the untrusted node (e.g. remove the folder and readd it) after upgrading to v1.16.0?  
There were bugs in v1.15.1 that are fixed, but the ramifications aren't cleaned up by those fixes - it just prevents the root cause.

**Comment by mklss** [comment]
I created a new share (i.e. removed and readded everywhere). Unfortunately, the problems persist, and are not solved by connecting only A<->U<->B either.
Basically, I quickly copied, renamed, deleted and renamed files like their original ones. Sometimes multiple copies, sometimes multiple rename steps. There were only two different files in the folder. Also, I renamed/deleted/copied on A and B. I don't think I got conflicts (wasn't fast enough and didn't try), or at least there's no indication.
Yet, I ended up with 3 files in A (one copy is alive), but only 2 files in B and U, and U and A tell me that A is out of sync, but I don't see what should be done to make A sync. It's "syncing 99%" with 0 or 1 B/s. (That is, A will neither delete nor sync its additional )copy, 

I tried similar things without encryption, and couldn't break it. At one point, it took a while, but eventually, it was always in sync.
